packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -130,3 +138,121 @@ postPublishedCallback.add(async (publishedPost: DbPost) => {
+  if (post.manifoldReviewMarketId) return;
+
+  const botUser = await context.Users.findOne({_id: reviewUserBot})
+  const annualReviewLink = 'https://www.lesswrong.com/tag/lesswrong-review'
+  const postLink = 'https://www.lesswrong.com/posts/' + post._id + '/' + post.slug
+
+  const year = post.postedAt.getFullYear()
+  const initialProb = 14
+  const question = `Will "${post.title.length < 50 ? post.title : (post.title.slice(0,45)+"...")}" make the top fifty posts in LessWrong's ${year} Annual Review?`
+  const descriptionMarkdown = `As part of LessWrong's [Annual Review](${annualReviewLink}), the community nominates, writes reviews, and votes on the most valuable posts. Posts are reviewable once they have been up for at least 12 months, and the ${year} Review resolves in February ${year+2}.\n\n\nThis market will resolve to 100% if the post [${post.title}](${postLink}) is one of the top fifty posts of the ${year} Review, and 0% otherwise. The market was initialized to ${initialProb}%.` // post.title
+  const closeTime = new Date(year + 2, 1, 1) // i.e. february 1st of the next next year (so if year is 2022, feb 1 of 2024)
+  const visibility = isProduction ? "public" : "unlisted"
+
+  if (!botUser) throw new Error("Bot user not found")
+
+  try {
+    const result = await fetch("https://api.manifold.markets/v0/market", {
+      method: "POST",
+      headers: {
+        authorization: `Key ${manifoldAPIKey}`,
+        "content-type": "application/json"
+      },
+      body: JSON.stringify({
+        outcomeType: "BINARY",
+        question: question,
+        descriptionMarkdown: descriptionMarkdown,
+        closeTime: Number(closeTime),
+        visibility: visibility,
+        // groupIds: groupIds,
+        initialProb: initialProb
+      })
+    })
+
+    const liteMarket = await result.json()
+
+    if (!result.ok) {
+      throw new Error(`HTTP error! status: ${result.status}`);
+    }
+
+    // update the database to include the market id
+    await Posts.rawUpdateOne(post._id, {$set: {manifoldReviewMarketId: liteMarket.id}})
+
+    const reviewTagSlug = 'annual-review-market';
+    const reviewYearTagSlug = `annual-review-${year}-market`;
+
+    // add the review tags to the post
+    await addTagToPost(post._id, reviewTagSlug, botUser, context);
+    await addTagToPost(post._id, reviewYearTagSlug, botUser, context);
+
+    // make a comment on the post with the market
+    const comment = await makeComment(post._id, year, liteMarket.url, botUser)
+    await Posts.rawUpdateOne(post._id, {$set: {annualReviewMarketCommentId: comment._id}})
+
+  } catch (error) {
+
+    //eslint-disable-next-line no-console
+    console.error('There was a problem with the fetch operation for creating a Manifold Market: ', error);
+  }
+})
+
+const makeComment = async (postId: string, year: number, marketUrl: string, botUser: DbUser) => {
```

Comment:
I think I'd give this a more descriptive name to indicate it's not making any random comment